rotkehlchen/db/evmtx.py

```diff
@@ -175,29 +175,16 @@ def get_evm_transactions(
         """
         query, bindings = filter_.prepare()
         if has_premium:
-            query = 'SELECT DISTINCT evm_transactions.tx_hash, evm_transactions.chain_id, timestamp, block_number, from_address, to_address, value, gas, gas_price, gas_used, input_data, nonce FROM evm_transactions ' + query  # noqa: E501
+            query = self._get_evm_transaction_query(query, has_premium)
```

Comment:
you are duplicating the if check here. Both before the function and then you do it also in the function.

The way around this is easy. Rename to `_form_evm_transaction_dbquery` and make it return a tuple of (str, list[tuple]) which means it will requrns the query string and its bindings.

Then you can avoid the external if.